src/frontend/mod.rs

```diff
@@ -146,6 +146,32 @@ impl<'a, B: Backend<'a>> Frontend<'a, B> {
         }
     }
 
+    fn convert_text(&mut self, mut text: Cow<'a, str>) {
+        // test if the text contains a latex command
+        lazy_static! {
+            // latex actually isn't really parsable, so this is only a very rough estimate of common commands
+            // Expect: \commandname<arg>[arg]{arg}{arg}...
+            static ref LATEX_COMMAND: Regex = Regex::new(r"\\[a-zA-Z]+(?:$|<.*?>|\[.*?\]|\{.*?\})").unwrap();
+        }
+        if let Some(m) = LATEX_COMMAND.find(&text) {
+            let (start, end, len) = (m.start(), m.end(), m.as_str().len());
+            if start != 0 {
+                let (l, r) = text.split_at(start);
+                text = r;
+                self.buffer.push_back(Event::Text(l));
+            }
+            if len != text.len() {
+                let (l, r) = text.split_at(end);
```

Comment:
Pretty sure this is the wrong index, `text` is the tail beginning from the regex match while `end` refers to the index in the original text. Avoiding split for empty slices is maybe not any faster either, surely not for slice and probably not for `String`, so rather something simple.

```
let (head, from_match) = text.split_at(start);
let (latex, tail) = tex.split_at(len);
if !head.is_empty() {
    self.buffer.push_back(Event::Text(head))
}
self.buffer.push_back(Event::Latex(latex));
// tail omitted, see next comment
```